### ISSUE TITLE
auto-update-config service doesn't check when the system was last updated

### ISSUE THREAD

**Issue by phillipzink** [mkellyxp/nixbook#8]
config.system.autoUpgrade only runs if the system is running at the time specified system.autoUpgrade.dates. Add an additional check for when the last generation was created and send a notification nudge to use the "Update and Reboot" or "Update and Shut Down" scripts to keep the system updated.

**Comment by mkellyxp** [comment]
Yes, good thought.  I guess my thinking was, if they're spending lots of time with the machine off and turnign it back on, they're probably getting these updates anyway.  My nudge in the code now is really aimed at people who just have their machines on constantly.

Do you know of a good way to get the date of the last known generation? that would be cool

**Comment by phillipzink** [comment]
I used nix-env --list-generations. I saw a different way using nixos-rebuild, but this ran faster.

[https://github.com/phillipzink/nixbook/blob/nixos-and-flatpak-update-fixes/base.nix](https://github.com/phillipzink/nixbook/blob/nixos-and-flatpak-update-fixes/base.nix)

**Comment by phillipzink** [comment]
> I guess my thinking was, if they're spending lots of time with the machine off and turnign it back on, they're probably getting these updates anyway.

That's only true if they're shutting down using the Upgrade and Shutdown scripts. If the machine is off at 4:40 on Mondays it never updates nixos. Flatpaks get upgraded either way. I think this is fine. I would rather nudge people to use the Upgrade and Shutdown scripts once a month than have nixos-rebuild start in the middle of a presentation.

**Comment by mkellyxp** [comment]
This has been tested and merged into main!  Thank you!
